image-download

```diff
@@ -279,12 +278,13 @@ def download_images(image_list, force, quiet, state, store):
                 target = committed_target(image)
 
             # don't download the same thing multiple times in parallel
-            lockfile = wait_lock(target)
+            lockfile, lock_fd = wait_lock(target)
 
             if force or state or not os.path.exists(target):
                 download(target, force, state, quiet, store)
 
             # clean up lock file; parallel runs may remove it already
+            lock_fd.close()
```

Comment:
Strictly speaking, that should live in a `finally:` block.

How about using the `@contextmanager` decorator with a coroutine and a `with:` statement?  I feel like that would be all-around safer, more foolproof, and also easier to read.

Reply:
Dropped that commit as you wanted to rework this.